### PR TITLE
Stories/project group management

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/views/project_groups/maintenance.haml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+- title = 'Project Group Maintenance'
```
Comment (awisema) [inline app/views/project_groups/maintenance.haml:4]: Add a breadcrumb at the top to go back to the project groups page?

File: app/views/project_groups/maintenance.haml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+= content_for(:title, title)
+%h1= content_for :title
+%p You can use this page to maintain project groups by uploading an Excel file.  Any project groups in the file that do not exist will be created, and all project groups included in the file will be updated to only contain the projects included in the file.  File the file should be an xlsx file and the format is as follows:
+.card.mb-2
+  %table.table.table-striped.mb-0
+    %thead
+      %th ProjectGroupName
+      %th ProjectName
```
Comment (awisema) [inline app/views/project_groups/maintenance.haml:12]: Names are just for readability and not used in the import?

File: app/views/project_groups/index.haml [hunk 3]
```diff
@@ -7,9 +7,13 @@
   .col-sm-8
     = render 'search_form'
   .col-sm-4.text-right
-    = link_to new_project_group_path, class: 'btn btn-primary' do
+    = link_to new_project_group_path, class: 'btn btn-primary mb-2' do
       %span.icon-plus
       Add Project Group
+    - if can_import_project_groups?
+      = link_to maintenance_project_groups_path, class: 'btn btn-secondary' do
```
Comment (awisema) [inline app/views/project_groups/index.haml]: `class: 'btn btn-secondary ml-2 mb-2'`

**Review by awisema — CHANGES_REQUESTED** [review]
If an import file would leave a project group empty, the behavior is confusing.

File: app/models/grda_warehouse/project_group.rb [hunk 4]
```diff
@@ -56,5 +57,71 @@ def self.options_for_select(user:)
...
+          [project_id, data_source_id],
+          id,
+        ]
+      end.to_h
+      all_project_keys = all_projects.keys
+
+      parsed.reject! { |row| row.values.map(&:blank?).any? }
+      parsed.select! { |row| all_project_keys.include?([row['ProjectID'].to_s, row['data_source_id']]) }
+      errors << "Excluded #{ActionController::Base.helpers.pluralize(input_count - parsed.count, 'project')}" if (input_count - parsed.count).positive?
+      return errors if parsed.empty?
```
Comment (eanders) [inline app/models/grda_warehouse/project_group.rb:86]: @awisema this should prevent leaving an existing project group empty.

**Comment by eanders** [comment]
@awisema this should be ready for a re-review, now with some simple test coverage.

**Pull request merged**